sysutils/pfSense-pkg-apcupsd/files/usr/local/www/widgets/widgets/apcupsd.widget.php

```diff
@@ -0,0 +1,416 @@
+			$user_settings["widgets"][$widgetkey]["apc_charge_critical_threshold"] = "15";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_bage_warning_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_bage_warning_threshold"] = "365";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_bage_critical_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_bage_critical_threshold"] = "720";
+		}
+		
+		if (!(include_once "/usr/local/pkg/apcupsd.inc")) {
+			$rtnstr = "";
+			$rtnstr .= "<div class=\"panel panel-warning responsive\"><div class=\"panel-heading\"><h2 class=\"panel-title\">apcupsd not installed...</h2></div>\n";
+			$rtnstr .= "<pre>\n";
+			$rtnstr .= "Please install and configure apcupsd before using this widget.<br />\n";
+			$rtnstr .= "</pre>\n";
+			$rtnstr .= "</div>\n";
+			print($rtnstr);
+			exit(1);
+		}
+		
+		if (check_nis_running_apcupsd()) {
+			$nisip = (check_nis_ip_apcupsd() != ''? check_nis_ip_apcupsd() : "localhost");
+			$nisport = (check_nis_port_apcupsd() != '' ? check_nis_port_apcupsd() : "3551");
+
+			$ph = popen("/usr/local/sbin/apcaccess -h " . escapeshellarg($nisip) . ":" . escapeshellarg($nisport) . " 2>&1", "r" );
+			while ($v = fgets($ph)) {
+				$results[trim(explode(': ',$v)[0])]=trim(explode(': ',$v)[1]);
+			}
+			pclose($ph);
+
+			$rtnstr = "<tr><td>Status</td><td colspan=\"3\">\n";
+			
+			if ($results != null) {
+				$bchrg = str_replace(" Percent", "", $results['BCHARGE']);
+				switch ($results['STATUS']) {
+					case 'ONLINE':
+						$rtnstr .= "<span class=\"fa fa-plug\" style=\"color:green;font-size:2em;transform: rotate(45deg);\"></span><span style=\"color:green;font-style:bold;font-size:2em\">\n";
+						break;
+					case 'CHARGING':
+						$bclr = "orange";
+						if ($bchrg <= 25) {
+							$bicn = "fa fa-battery-empty";
+						} elseif ($bchrg <= 50) {
+							$bicn = "fa fa-battery-quarter";
+						} elseif ($bchrg <= 75) {
+							$bicn = "fa fa-battery-half";
+						} elseif ($bchrg <= 99) {
+							$bicn = "fa fa-battery-three-quarters";
+						} else {
+							$bicn = "fa fa-battery-empty";
+						}
+						$rtnstr .= "<span class=\"" . $bicn . "\" style=\"color:" . $bclr . ";font-size:2em;transform: rotate(270deg);\"></span><span style=\"color:" . $bclr . ";font-style:bold;font-size:2em\">\n";
+						break;
+					default:
+					case 'ONBATT':
+						$bclr = "red";
+						if ($bchrg <= 25) {
+							$bicn = "fa fa-battery-empty";
+						} elseif ($bchrg <= 50) {
+							$bicn = "fa fa-battery-quarter";
+						} elseif ($bchrg <= 75) {
+							$bicn = "fa fa-battery-half";
+						} elseif ($bchrg <= 99) {
+							$bicn = "fa fa-battery-three-quarters";
+						} else {
+							$bicn = "fa fa-battery-full";
+						}
+						$rtnstr .= "<span class=\"" . $bicn . "\" style=\"color:" . $bclr . ";font-size:2em;transform: rotate(270deg);\"></span><span style=\"color:" . $bclr . ";font-style:bold;font-size:2em\">\n";
+						break;
+				}
```

Comment:
I was thinking this entire block of code could be simplified as (UNTESTED):

```php
if ($results['STATUS'] == 'ONLINE') {
	$bicn = 'fa fa-plug';
	$bclr = 'green';
} else {
	if ($results['STATUS'] == 'CHARGING') {
		$bclr = 'orange';
	} else {
		$bclr = 'red';
	}

	$bchrg = str_replace(" Percent", "", $results['BCHARGE']);
	if ($bchrg <= 25) {
		$bicn = "fa fa-battery-empty";
	} elseif ($bchrg <= 50) {
		$bicn = "fa fa-battery-quarter";
	} elseif ($bchrg <= 75) {
		$bicn = "fa fa-battery-half";
	} elseif ($bchrg <= 99) {
		$bicn = "fa fa-battery-three-quarters";
	} else {
		$bicn = "fa fa-battery-empty";
	}
}

$rtnstr .= "<span class=\"" . $bicn . "\" style=\"color:" . $bclr . ";font-size:2em;transform: rotate(270deg);\"></span><span style=\"color:" . $bclr . ";font-style:bold;font-size:2em\">\n";
```

Reply:
Changes made